ocaml/xapi-idl/lib/debuginfo.ml

```diff
@@ -47,3 +47,21 @@ let to_string t =
 
 (* Used for xapi-idl servers that do not yet accept tracing data in the debuginfo *)
 let to_log_string t = t.log
+
+(* Sets the logging context based on `dbg`.
+   Also adds a new tracing span, linked to the parent span from `dbg`, if available. *)
+let with_dbg ?(with_thread = false) ?(dbg = "") ~module_name ~name f =
```

Comment:
Why is `dbg` and optional argument? It seems odd to have a function called `with_dbg` that doesn't require a `dbg`. And the empty string doesn't help.

Reply:
Correction... the function is called `with_dbg` because _provides_ a `dbg` to the function. Still, it would be better to make `~dbg` mandatory, since this is the thing that links the spans.